osu.Game/Screens/Select/Carousel/CarouselRoot.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+namespace osu.Game.Screens.Select.Carousel
+{
+    public class CarouselRoot : CarouselGroupEagerSelect
+    {
+        private readonly BeatmapCarousel carousel;
+
+        public CarouselRoot(BeatmapCarousel carousel)

```

Comment:
probably want to pass the function here as a delegate, this is introducing outwards dependencies which we want to avoid.

Reply:
That's the reason why I went down the path of `CarouselRoot`. It's an alternative to delegate that serves the same purpose.

Reply:
as in i still don't want to see `BeatmapCarousel` passed in, but `Action selectionFunction` instead.